### PR TITLE
[App Service] `az webapp up`: Fix `--logs` arg fails with an exception

### PR DESCRIPTION
**Pull request by sarsharma**
**Related command**

`az webapp up --logs`

**Description**

The command `az webapp up` fails if the `--logs` arg is passed. This is because logstreaming does not start working until the main site has been started and therefore, we get an exception while trying to stream logs. To fix this, we can ping the site to warmup prior to starting the logstreaming thread. 
Related issue: https://github.com/Azure/azure-cli/issues/23058

**History Notes**

[App Service] `az webapp up`: Fix `--logs` arg fails with an exception

**Testing Guide**

Test deploying webapps using `az webapp up --logs` 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 from nacl import encoding, public
 
 import OpenSSL.crypto
+import urllib3
```
Comment (sgollapudi77) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:27]: Do we need to add this library in requirements.txt somewhere?

Comment (sarsharma) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:27]: no, this is already in use today

**Review by shubham805 — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -4841,7 +4843,17 @@ def webapp_up(cmd, name=None, resource_group_name=None, plan=None, location=None
 
     if logs:
         _configure_default_logging(cmd, rg_name, name)
-        return get_streaming_log(cmd, rg_name, name)
+        # ping the site so that the site container is ready
+        # logsteam does not work if the site container has not started
+        # required for https://github.com/Azure/azure-cli/issues/23058
+        _url = _get_url(cmd, rg_name, name)
+        logger.warning("Waiting for the app to be ready ...")
```
Comment (vivekjilla) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:4850]: why is this a warning? seems like an info to me

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -4841,7 +4843,17 @@ def webapp_up(cmd, name=None, resource_group_name=None, plan=None, location=None
 
     if logs:
         _configure_default_logging(cmd, rg_name, name)
-        return get_streaming_log(cmd, rg_name, name)
+        # ping the site so that the site container is ready
+        # logsteam does not work if the site container has not started
+        # required for https://github.com/Azure/azure-cli/issues/23058
+        _url = _get_url(cmd, rg_name, name)
```
Comment (vivekjilla) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:4849]: we can move the `_url = ` and `http = ` also inside the `try` no?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (sarsharma) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: only warning logs get printed by default for the user. This seems to be the convention for the repo.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 3 above)
Comment (sarsharma) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: updated, thanks!

**Review by kumaramit-msft — APPROVED** [review]

**Comment by sarsharma** [comment]
Seems like the CI is not picking up the updated PR title and hence the PR Title Format test is failing. Can you please take a look @yonzhan

**Comment by vivekjilla** [comment]
lgtm!

**Review by vivekjilla — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by sarsharma** [comment]
@yonzhan, the failing CI test seems to be an unrelated issue.

**Comment by sarsharma** [comment]
@jsntcy, @zhoxing-ms, @yanzhudd
Pinging for review

**Comment by yanzhudd** [comment]
Please add test for this change to check if it meets your expectations.

**Comment by zhoxing-ms** [comment]
Please note that we are launching the release for this sprint this week. Please resolve all comments and conflicts by Tomorrow, otherwise the release of this PR will have to be postponed to the next sprint **(on 12-05)**

**Comment by sarsharma** [comment]
Hi @yanzhudd, @zhoxing-ms 
I have updated the PR with tests and resolved all comments. Please review.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**